depthai_ros_driver/include/depthai_ros_driver/device_ros.hpp

```diff
@@ -74,39 +85,50 @@ class DeviceROS : public dai::DeviceBase {
 protected:
     // create stream based on name at the time of subscription
     template <class MsgType>
-    auto generate_cb_lambda(std::string name) {
-        auto conn = this->getConnection();
-        const auto core_sub_lambda = [&, stream = dai::XLinkStream{*conn, name, dai::XLINK_USB_BUFFER_MAX_SIZE}](
-                                             MsgType& msg, dai::RawBuffer& data) {
+    auto generate_cb_lambda(std::unique_ptr<dai::XLinkStream>& stream) -> boost::function<void(const boost::shared_ptr<MsgType const >&)> {
+        // auto conn = this->getConnection();
+        // const auto core_sub_lambda = [stream = dai::XLinkStream{*conn, name, dai::XLINK_USB_BUFFER_MAX_SIZE}](
+        const auto core_sub_lambda = [&stream] (
+                                             const boost::shared_ptr<MsgType const >& msg) {
             Guard guard([] { ROS_ERROR("Communication failed: Device error or misconfiguration."); });
 
-            // TODO:  convert msg to data here somehow
-            std::vector<std::uint8_t> serialized = dai::StreamPacketParser::serializeMessage(data);
-            stream.write(serialized);
+            // convert msg to data
+            msgpack::sbuffer sbuf;
```

Comment:
Can we have a vector of sbuffer and use a reference here (without using a map)? It'd make this lambda allocation free